integrated_channels/sap_success_factors/transmitters/learner_data.py

```diff
@@ -28,5 +32,13 @@ class SuccessFactorsLearnerDataTransmitter(SuccessFactorsTransmitterBase):
     (taken from OCNWebServicesConfiguration and OCNWebServicesEnterpriseCustomerConfiguration) and the oauth token.
 
     Record information about the success/failure of posting the learner data to SuccessFactors in
-     CompletionStatusEventAudit
+    the LearnerDataTransmissionAudit object provided, and save it to the database.
+
     """
+    def transmit(self, learner_data):
+        """
+        Transmit the learner data payload to the SAP SuccessFactors channel.
+        """
+        # TODO ENT-221 will handle transmitting the learner data payload to the channel
+        # Question: do we only want to transmit if learner_data.course_completed == True?
```

Comment:
Sounds good to me.

> The case we are not handling is audit/self-paced learners, but this might end up being a follow up item at this point.

A follow-up item would be great.  We can use the Grades API to fetch the current grade, but we need to know what the cut-off point is, especially since we can only send once per enrolment.

> It is unclear to me from reading through this PR if the LearnerDataTransmissionAudit object is getting persisted.
The `LearnerDataTransmissionAudit` object is currently only being created, not persisted.

> 1. Determine if an event was already sent for this learner/course, referencing the audit table.
Yep, uniqueness can be easily done with the `enterprise_course_enrollment_id`.  It sounds like we also want to check that the `course_completed == True` before saving.

> 2. Perform the call to SuccessFactors if this is a new event, else exit.
Yes.

> 3. Save a new audit object reflecting the status of the call.
Yep.  Can use the `status` and `error_message` fields - I wasn't sure if [`status` was supposed to come from the `certificate` or not?](https://github.com/edx/edx-enterprise/pull/59/files#diff-b2d1ad7443d2344126a60d0213e2d192R148).  I can fix that if it's intended for the SAP response.

I assume this will be part of a separate PR, after #59 and #65 merge?